include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -161,24 +161,24 @@ struct data_converter<CArray,
 };
 
 // Generic container converter
-template <typename Container, typename T = typename inspector<Container>::base_type>
+template <typename Container, typename T = typename inspector<Container>::element_type>
 struct container_converter {
-    typedef T value_type;
+    using value_type = T;
 
     inline container_converter(const DataSpace& space)
         : _space(space) {}
 
-    // Ship (pseudo)1D implementation
     inline value_type* transform_read(Container& vec) const {
         auto&& dims = _space.getDimensions();
-        if (!is_1D(dims))
-            throw DataSpaceException("Dataset cant be converted to 1D");
-        vec.resize(compute_total_size(dims));
-        return vec.data();
+        if (is_1D(dims)) { // This is a bad things, but keep for compatibility
+            dims = std::vector<size_t>{compute_total_size(dims)};
+        }
+        inspector<Container>::resize(vec, dims);
+        return inspector<Container>::first(vec);
     }
 
     inline const value_type* transform_write(const Container& vec) const noexcept {
-        return vec.data();
+        return inspector<Container>::first(vec);
```

Comment:
Just to understand what you are doing, is this a free function that mimics e.g. `vec.begin()` such that it can be done for all types (also for those that do not provide such a method)?

A remark that pops up now that I see this code is that the names `Container` and `vec` (together) confuse me a bit. Wouldn't `inspector<T>::begin(container)` or `inspector<T>::begin(data)` be more descriptive? Or did I misunderstood what this code does?

Reply:
begin return an iterator.

The reason is to be generic for boost ublas matrix, this is implemented as `matrix.data(0, 0)`

Reply:
I would still call `first` actually `begin` to conform more to typical naming...?